### ISSUE TITLE
No response - No device - Maybe fix

### ISSUE THREAD

**Issue by xinteksik** [YinHangCode/homebridge-mi-aqara#259]
Problem is in index.js:

var command1 = '{"cmd":"read", "sid":"' + gatewaySid + '"}';

This command doesnt work on gateway. Than no version is set and nothing works.

I fix it by adding this after line 454:
proto_version: jsonObj['proto_version'],
model: jsonObj['model']

I know, some bugs is still there, but it works :-)

**Comment by studentIvan** [comment]
This hack works like magic, my hack for Gateway Security Kit:

```js
var gateway = that.GatewayUtil.getBySid(gatewaySid);
        if(!gateway) {
            console.log(jsonObj);
            console.log('now gateway motherfucker');
            // add gateway
            gateway = {
                sid: gatewaySid,
                passwd: that.ConfigUtil.getGatewayPasswordByGatewaySid(gatewaySid),
                ip: jsonObj['ip'], // rinfo.address,
                port: jsonObj['port'], // rinfo.port,
                proto_version: '1.1.2', // hardcoded
                // proto_version: jsonObj['proto_version'], // this field was empty
                // model: jsonObj['model'] // this field was empty
                model: 'gateway' // hardcoded
            }
            console.log('add this gateway', gateway);
            gateway = that.GatewayUtil.addOrUpdate(gatewaySid, gateway);
        }
```

before line `if(that.ConfigUtil.isConfigGateway(gatewaySid)) {`

there were no proto_version and model and I've simple hardcoded it hehe, now manage show me the devices, at least this is the progress

![image](https://user-images.githubusercontent.com/1205208/51072572-fe006680-1673-11e9-98e6-c417f46b20cb.png)

**Comment by studentIvan** [comment]
`passwd: undefined,` // should be filled with your gateway password

**Comment by marcoschmidt79** [comment]
> `passwd: undefined,` // should be filled with your gateway password

Yes. This gateway is configured in a different instance. my problem concerns the gateway with the ip 127.

an 22 22:17:38 homebridge homebridge[11763]: { cmd: 'iam',
Jan 22 22:17:38 homebridge homebridge[11763]: port: '9898',
Jan 22 22:17:38 homebridge homebridge[11763]: sid: '04cf8c8fa58b',
Jan 22 22:17:38 homebridge homebridge[11763]: model: 'gateway',
Jan 22 22:17:38 homebridge homebridge[11763]: proto_version: '1.1.2',
Jan 22 22:17:38 homebridge homebridge[11763]: ip: '192.168.5.127' }
Jan 22 22:17:38 homebridge homebridge[11763]: now gateway motherfucker
Jan 22 22:17:38 homebridge homebridge[11763]: add this gateway { sid: '04cf8c8fa58b',
Jan 22 22:17:38 homebridge homebridge[11763]: passwd: 'xxxxxxxxxxxxxxxxxxx',

**Comment by studentIvan** [comment]
> > `passwd: undefined,` // should be filled with your gateway password
> 
> Yes. This gateway is configured in a different instance. my problem concerns the gateway with the ip 127.
> 
> an 22 22:17:38 homebridge homebridge[11763]: { cmd: 'iam',
> Jan 22 22:17:38 homebridge homebridge[11763]: port: '9898',
> Jan 22 22:17:38 homebridge homebridge[11763]: sid: '04cf8c8fa58b',
> Jan 22 22:17:38 homebridge homebridge[11763]: model: 'gateway',
> Jan 22 22:17:38 homebridge homebridge[11763]: proto_version: '1.1.2',
> Jan 22 22:17:38 homebridge homebridge[11763]: ip: '192.168.5.127' }
> Jan 22 22:17:38 homebridge homebridge[11763]: now gateway motherfucker
> Jan 22 22:17:38 homebridge homebridge[11763]: add this gateway { sid: '04cf8c8fa58b',
> Jan 22 22:17:38 homebridge homebridge[11763]: passwd: 'xxxxxxxxxxxxxxxxxxx',

does the miio see your device? also maybe this message can help you https://github.com/aholstenson/miio/issues/92#issuecomment-453737136

**Comment by ghdbwise** [comment]
After some further testing i found out that when storing the token from the Mi Home App in /root/.local/share/miio/tokens.json - miio discover will see my gateway and devices

pi@rpi21:~ $ miio discover
 INFO  Discovering devices. Press Ctrl+C to stop.

Device ID: 129600562
Model info: Unknown
Address: 192.168.101.62
Token: ???
Support: Unknown

Device ID: 253569859
Model info: Unknown
Address: 192.168.101.67
Token: ???
Support: Unknown

Device ID: 131869114
Model info: lumi.gateway.v3
Address: 192.168.101.75
Token: a3709927c2622f13e25c8a20ec39655b via stored token
Support: At least basic

Device ID: 158d000204401e
Model info: lumi.switch.v2
Address: Owned by miio:131869114
Token: Automatic via parent device
Support: At least basic

Device ID: 158d000237ca97
Model info: lumi.magnet.aq2
Address: Owned by miio:131869114
Token: Automatic via parent device
Support: At least basic

Device ID: b57fffe13f4d0
Model info: lumi.generic.85
Address: Owned by miio:131869114
Token: Automatic via parent device
Support: At least basic

and mio inspect <ip> will further information:
pi@rpi21:~ $ miio inspect 192.168.101.75
 INFO  Attempting to inspect 192.168.101.75

Device ID: 131869114
Model info: lumi.gateway.v3
Address: 192.168.101.75
Token: a3709927c2622f13e25c8a20ec39655b via stored token
Support: At least basic

Type info: sensor, miio:gateway, miio
Capabilities: illuminance, state, children

Firmware version: 1.4.1_167
Hardware version: MW300
MCU firmware version: 0158

WiFi: PITS!Box 6490 (CC:CE:1E:EE:45:E9) RSSI: -68
WiFi firmware version: SD878x-14.76.36.p84-702.1.0-WM

Remote access (Mi Home App): UDP

Properties:
  - illuminance: 397
  - rgb: [object Object]
  - brightness: 0

But at least Homebridge still has no Mi Gateway and/or Devices.

**Comment by a1cellsolution** [comment]
Hey guys i am a total noob got the xiaomi hub working with the homekit few months before , but not working , the hub is working as security device using the other plugin , but not with this plugin . can someone please post a step by step fix how to apply this fix 

thanks
